scheduler/scheduler_test.go

```diff
@@ -0,0 +1,155 @@
+package scheduler
+
+import (
+	"bytes"
+	"log/slog"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func TestScheduler(t *testing.T) {
+	t.Run("test type", func(t *testing.T) {
+		scheduler1, _ := NewTaskSchedule(5, 40)
+		got := reflect.TypeOf(scheduler1)
+		expected := reflect.TypeOf(&TaskSchedule{})
+		if got != expected {
+			t.Errorf("got %v; expected %v", got, expected)
+		}
+	})
+	t.Run("test type error", func(t *testing.T) {
+		scheduler1, _ := NewTaskSchedule(5, 40)
+		got := reflect.TypeOf(scheduler1)
+		notExpected := reflect.TypeOf(&taskHandler{})
+		if got == notExpected {
+			t.Errorf("got %v; didnt expect %v", got, notExpected)
+		}
+	})
+}
+func TestWithRetry(t *testing.T) {
+	const retryNum = 4
+	t.Run("test with retry", func(t *testing.T) {
+		ts, _ := NewTaskSchedule(10, 5)
+		ts.Register(&MyTask{}, WithRetry(4))
+		ts.Start()
+		ts.Stop()
+		got := ts.doneRun.Load()
+
+		if got != retryNum {
+			t.Errorf("got %v; expected %v", got, retryNum)
+		}
+	})
+}
+func TestWithName(t *testing.T) {
+	t.Run("test with name", func(t *testing.T) {
+		name := "the best task"
+		ts, _ := NewTaskSchedule(10, 3)
+		exampleTask := MyTask{}
+		ts.Register(&exampleTask, WithName(name))
+		ts.Start()
+		ts.Stop()
+		results := ts.getResults()
+		if len(results) == 0 {
+			t.Errorf("got %v; expected not %v", results, results)
+		}
+		if results[0].task.name != name {
+			t.Errorf("got %v; expected %v", results[0].task.name, name)
+		}
+	})
+}
+
+func createTestLogger() (*slog.Logger, *bytes.Buffer) {
+	buf := &bytes.Buffer{}
+	var logger *slog.Logger
+	logger = slog.New(slog.NewTextHandler(buf, nil))
+	return logger, buf
+}
+
+func TestWithLogging(t *testing.T) {
+	tests := []struct {
+		name              string
+		shouldFindMessage string
+		shouldFind        bool
+	}{
+		{
+			name:              "captures messages exist",
+			shouldFindMessage: "registered",
+			shouldFind:        true,
+		},
+		{
+			name:              "captures messages not exist",
+			shouldFindMessage: "banana",
+			shouldFind:        false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testLogger, buf := createTestLogger()
+			ts, _ := NewTaskSchedule(10, 4)
+			exampleTask := &MyTask{}
+			ts.Register(exampleTask, WithLogging(testLogger))
+			ts.Start()
+			ts.Stop()
+
+			output := buf.String()
+			hasMessage := strings.Contains(output, tt.shouldFindMessage)
+
+			if tt.shouldFind && !hasMessage {
+				t.Errorf("Expected to find '%s' in logs, got: %s",
+					tt.shouldFindMessage, output)
+			} else if !tt.shouldFind && hasMessage {
+				t.Errorf("Did not expect to find '%s' in logs, got: %s",
+					tt.shouldFindMessage, output)
+			}
+		})
+	}
+}
+
+// Example demonstrates the creation, registration, execution, and summary of tasks using the TaskSchedule infrastructure.
+// Dropping the time key provides us consistent output.
+func Example() {
```

Comment:
Please make sure the give indicative name for the example tests.
```suggestion
func ExampleScheduler() {
```

Reply:
Fixed!